### PR TITLE
Supporting UUID column type in shardingsphere

### PR DESCRIPTION
**Pull request by cb-mukulsharma**
Fixes [#24043](https://github.com/apache/shardingsphere/issues/24043).

Changes proposed in this pull request:
  - Supporting the queries have UUID column type in Postgres database with ShardingSphere

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValue.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+package org.apache.shardingsphere.db.protocol.postgresql.packet.command.query.extended.bind.protocol;
+
+import org.apache.shardingsphere.db.protocol.postgresql.payload.PostgreSQLPacketPayload;
+import org.postgresql.util.ByteConverter;
+
+import java.util.UUID;
+
+public class PostgreSQLUUIDBinaryProtocolValue implements PostgreSQLBinaryProtocolValue {
```
Comment (strongduanmu) [inline db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValue.java:25]: Please add final and javadoc for PostgreSQLUUIDBinaryProtocolValue.

File: db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValue.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return new UUID(ByteConverter.int8(result, 0), ByteConverter.int8(result, 8));
+    }
+    
+    @Override
+    public void write(final PostgreSQLPacketPayload payload, final Object value) {
+        UUID uuid = (UUID) value;
+        payload.writeInt8(uuid.getMostSignificantBits());
+        payload.writeInt8(uuid.getLeastSignificantBits());
+    }
+
```
Comment (strongduanmu) [inline db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValue.java:45]: Please remove useless blank line.

File: db-protocol/postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValueTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    
+    @Test
+    void assertRead() {
+        UUID randomUUID = UUID.randomUUID();
+        byte[] expected = new byte[16];
+        ByteBuffer buffer = ByteBuffer.wrap(expected);
+        buffer.putLong(randomUUID.getMostSignificantBits());
+        buffer.putLong(randomUUID.getLeastSignificantBits());
+
```
Comment (strongduanmu) [inline db-protocol/postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValueTest.java:46]: Please remove useless blank line.

File: db-protocol/postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValueTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    
+    @Test
+    void assertWrite() {
+        UUID randomUUID = UUID.randomUUID();
+        byte[] expected = new byte[16];
+        ByteBuffer buffer = ByteBuffer.wrap(expected);
+        buffer.putLong(randomUUID.getMostSignificantBits());
+        buffer.putLong(randomUUID.getLeastSignificantBits());
+
```
Comment (strongduanmu) [inline db-protocol/postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValueTest.java:59]: Please remove useless blank line.

File: db-protocol/postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValueTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    void assertWrite() {
+        UUID randomUUID = UUID.randomUUID();
+        byte[] expected = new byte[16];
+        ByteBuffer buffer = ByteBuffer.wrap(expected);
+        buffer.putLong(randomUUID.getMostSignificantBits());
+        buffer.putLong(randomUUID.getLeastSignificantBits());
+        
+        ByteBuf byteBuf = Unpooled.wrappedBuffer(expected);
+        PostgreSQLPacketPayload payload = new PostgreSQLPacketPayload(byteBuf.writerIndex(0), StandardCharsets.UTF_8);
+
```
Comment (strongduanmu) [inline db-protocol/postgresql/src/test/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/extended/bind/protocol/PostgreSQLUUIDBinaryProtocolValueTest.java:62]: Please remove useless blank line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]
Hi @cb-mukulsharma, can you add e2e test for uuid type?

**Comment by cb-mukulsharma** [comment]
> Hi @cb-mukulsharma, can you add e2e test for uuid type?
Hi @strongduanmu, can you please share some example PR or class that can help to write e2e tests?

**Comment by cb-mukulsharma** [comment]
> Hi @cb-mukulsharma, can you add e2e test for uuid type?


> Hi @cb-mukulsharma, can you add e2e test for uuid type?

**Comment by cb-mukulsharma** [comment]
> Hi @cb-mukulsharma, can you add e2e test for uuid type?

Hi @strongduanmu, Sure e2e tests for uuid added.

**Comment by strongduanmu** [comment]
> > Hi @cb-mukulsharma, can you add e2e test for uuid type?
> 
> Hi @strongduanmu, Sure e2e tests for uuid added.

Good job, I will review this pr again.

**Comment by strongduanmu** [comment]
Hi @cb-mukulsharma, can you run e2e test on your local machine?

**Comment by cb-mukulsharma** [comment]
> Hi @cb-mukulsharma, can you run e2e test on your local machine?

Hi @strongduanmu, E2E tests are running successful. Here is the test report:
Screenshot 2024-01-09 at 3 16 43 PM

**Comment by strongduanmu** [comment]
> > Hi @cb-mukulsharma, can you run e2e test on your local machine?
> 
> Hi @strongduanmu, E2E tests are running successful. Here is the test report: Screenshot 2024-01-09 at 3 16 43 PM

@cb-mukulsharma Can you take a look for https://github.com/apache/shardingsphere/actions/runs/7443816543/job/20296126253#step:7:14882

**Comment by cb-mukulsharma** [comment]
> > > Hi @cb-mukulsharma, can you run e2e test on your local machine?
> > 
> > 
> > Hi @strongduanmu, E2E tests are running successful. Here is the test report: Screenshot 2024-01-09 at 3 16 43 PM
> 
> @cb-mukulsharma Can you take a look for https://github.com/apache/shardingsphere/actions/runs/7443816543/job/20296126253#step:7:14882

Hey @strongduanmu, I have fixed the issue with test case. It is working fine in local. Can you please re-run the E2E test-suites. Thanks.

**Review by strongduanmu — APPROVED** [review]
LGTM, @RaigorJiang Can you help check this pr?

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
Great job, thank you! @cb-mukulsharma @strongduanmu

**Pull request merged**